src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_POSITION = 0;
+    private static final int NAME_POSITION = 1;
+    private static final int HOURS_POSITION = 2;
+    private static final int RATE_POSITION = 3;
+    private static final String DELIMITER = " - ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateStart = null;
+        try {
+            dateStart = LocalDate.parse(dateFrom, FORMATTER);
+        } catch (DateTimeParseException e) {
+            throw new RuntimeException("Start date is incorrect: " + dateFrom);
+        }
```

Comment:
```suggestion
  private LocalDate parseLocalDate(String date) {
        try {
            return LocalDate.parse(date, FORMATTER);
        } catch (DateTimeParseException e) {
            throw new RuntimeException("Can't parse date: " + date);
        }
    }
```
Create such private method and simplify your solution ;)
The same for integer values